### ISSUE TITLE
ModelCopyManager consumes an ever increasing amount of memory over time

### ISSUE THREAD

**Issue by goneall** [spdx/Spdx-Java-Library#260]
As reported by #258 - the `ModelCopyManager` consumes an ever increasing amount of memory due to caching the copied ID's.

One possible solution would be to implement some form of garbage collection.  When a `ModelStore` is destroyed, it could call back to the `ModelCopyManager` to remove all the associated references from the cache.

Note that this approach may have some CPU performance implications due to having to lock the cache during the operation.

**Comment by martin-bpw** [comment]
Yes this would also be an option, I was already considering doing something like this (but using reflection to access the state).

**Comment by goneall** [comment]
This is a rather significant changes, targeting for the 2.1 release
